src/datatable.py

```diff
@@ -0,0 +1,192 @@
+
+        self.run_queries()
+
+    def output_result_on_given_fields(self):
+        output = {}
+        output['draw'] = str(int(self.request['draw']))
+        output['recordsTotal'] = str(self.records_total)
+        output['recordsFiltered'] = str(self.records_filtered)
+
+        data_rows = []
+        i = self.paging().start
+
+        for row in self.result_data:
+            data_row = []
+            i += 1
+            data_row.append("<a href='/vus/{0}'>{1}</a>".format(row['_id'], i))
+            for col in self.columns:
+                if col in row:
+                    val = row[col]
+                else:
+                    val = "-"
+
+                if isinstance(val, dict):
+                    print("YAS")
+                    val = json.dumps(val)
+                    print(val)
+                elif isinstance(val, list):
+                    val = ", ".join(val)
+
+                if isinstance(val, str):
+                    val = (val[:max_string_length] + '...') if len(val) > max_string_length else val
+
+                uwu = ""
+                if col == "fullgnomen":
+                    uwu = '<a href="/variant/{0}">{0}</a>'.format(val)
+                elif col == "dn_no":
+                    uwu = '<a href="/patient/{0}">{0}</a>'.format(val)
+                elif col == "gene":
+                    uwu = '<a href="/gene/{0}">{0}</a>'.format(val)
+                else:
+                    uwu = val
+                data_row.append(uwu)
+            data_rows.append(data_row)
+        output['data'] = data_rows
+        return output
+
+    def output_result_on_queried_fields(self):
+        output = {}
+        output['draw'] = str(int(self.request['draw']))
+        output['recordsTotal'] = str(self.records_total)
+        output['recordsFiltered'] = str(self.records_filtered)
+
+        data_rows = []
+        i = self.paging().start
+        for row in self.result_data:
+            data_row = []
+            i += 1
+            data_row.append(i)
+            for col, val in row.items():
+                uwu = ""
+                if col == "_id":
+                    if self.group_by == "fullgnomen":
+                        uwu = '<a href="/variant/{0}">{0}</a>'.format(val)
+                    elif self.group_by == "dn_no":
+                        uwu = '<a href="/patient/{0}">{0}</a>'.format(val)
+                    elif self.group_by == "gene":
+                        uwu = '<a href="/gene/{0}">{0}</a>'.format(val)
+                elif col == "total":
+                    uwu = val
+                else:
+                    # filtering like this doesn't work yet
+                    # uwu = '<a href="/all?{1}={0}">{0}</a>'.format(val, col)
+                    uwu = val
+                data_row.append(uwu)
+            data_rows.append(data_row)
+
+        output['data'] = data_rows
+        return output
+
+    def run_queries(self):
```

Comment:
When is this function called / executed? For any type of query or are there specific queries or a generic deliverable?
Based on the code, I get the impression it is more of a filter and grouping function than performing a query. 
I wonder if the function title is specific enough and matches with its functionality.

Reply:
I think this code runs queries to the mongodb database, to retrieve information that will show in the datatable, and then filtered/sorted based on the settings provided.
So for example if I want to use the search bar it needs to change the totals.

The filtering is done with another function called `filtering()`. which is called as one of the first lines in `run_queries()`.

It is a generic deliverable, as any datatable in the VUSualizer is made based on the information that needs to be shown at that time. So although the function name is a bit vague, I also think that this is due to the broadness of this whole file (datatable.py). It takes information (for example dictionaries), but you do not know beforehand what it is, because it is a very generic script that is used multiple times for all datatables.

I left the name as is, but if you have a good suggestion I can change it.